statsmodels/base/tests/test_distributed_estimation.py

```diff
@@ -0,0 +1,178 @@
+
+def test_gen_gamma_hat():
+
+    X = np.random.normal(size=(50, 3))
+    y = np.random.randint(0, 2, size=50)
+    beta = np.random.normal(size=3)
+    mod = OLS(y, X)
+    ghat = _gen_gamma_hat(X, 0, 3, 50, 0.01)
+    assert_equal(ghat.shape, (2,))
+
+
+def test_gen_tau_hat():
+
+    X = np.random.normal(size=(50, 3))
+    y = np.random.randint(0, 2, size=50)
+    beta = np.random.normal(size=3)
+    mod = OLS(y, X)
+    ghat = np.random.normal(size=2)
+    that = _gen_tau_hat(X, ghat, 0, 3, 50, 0.01)
+    assert_(isinstance(that, float))
+
+
+def test_gen_theta_hat():
+
+    X = np.random.normal(size=(50, 3))
+    y = np.random.randint(0, 2, size=50)
+    beta = np.random.normal(size=3)
+    mod = OLS(y, X)
+    ghat_l = []
+    that_l = []
+    for i in range(3):
+        ghat = _gen_gamma_hat(X, i, 3, 50, 0.01)
+        that = _gen_tau_hat(X, ghat, i, 3, 50, 0.01)
+        ghat_l.append(ghat)
+        that_l.append(that)
+    theta_hat = _gen_theta_hat(ghat_l, that_l, 3)
+    assert_equal(theta_hat.shape, np.eye(3).shape)
+
+
+def test_est_regularized_distributed():
+
+    X = np.random.normal(size=(50, 3))
+    y = np.random.randint(0, 2, size=50)
+    beta = np.random.normal(size=3)
+    mod = OLS(y, X)
+    res = _est_regularized_distributed(mod, 0, 2, fit_kwds={"alpha": 0.5})
+    bhat = res[0]
+    grad = res[1]
+    ghat_l = res[2]
+    that_l = res[3]
+
+    assert_(isinstance(res, tuple))
+    assert_equal(bhat.shape, beta.shape)
+    assert_equal(grad.shape, beta.shape)
+    assert_(isinstance(ghat_l, list))
+    assert_(isinstance(that_l, list))
+    assert_equal(len(ghat_l), len(that_l))
+    assert_equal(ghat_l[0].shape, (2,))
+    assert_(isinstance(that_l[0], float))
+
+    mod = GLM(y, X, family=Binomial())
+    res = _est_regularized_distributed(mod, 0, 2, fit_kwds={"alpha": 0.5})
+    bhat = res[0]
+    grad = res[1]
+    ghat_l = res[2]
+    that_l = res[3]
+
+    assert_(isinstance(res, tuple))
+    assert_equal(bhat.shape, beta.shape)
+    assert_equal(grad.shape, beta.shape)
+    assert_(isinstance(ghat_l, list))
+    assert_(isinstance(that_l, list))
+    assert_equal(len(ghat_l), len(that_l))
+    assert_equal(ghat_l[0].shape, (2,))
+    assert_(isinstance(that_l[0], float))
+
+
+def test_join_debiased():
+
+    X = np.random.normal(size=(50, 3))
+    y = np.random.randint(0, 2, size=50)
+    beta = np.random.normal(size=3)
+    mod = OLS(y, X)
+    res_l = []
+    for i in range(2):
+        res = _est_regularized_distributed(mod, i, 2, fit_kwds={"alpha": 0.5})
+        res_l.append(res)
+    joined = _join_debiased(res_l, 2)
+    assert_equal(joined.shape, beta.shape)
+
+    mod = GLM(y, X, family=Binomial())
+    res_l = []
+    for i in range(2):
+        res = _est_regularized_distributed(mod, i, 2, fit_kwds={"alpha": 0.5})
+        res_l.append(res)
+    joined = _join_debiased(res_l, 2)
+    assert_equal(joined.shape, beta.shape)
+
+
+def test_distributed_estimation():
+
+    X = np.random.normal(size=(50, 3))
```

Comment:
Use np.random.seed(34341) [the number is arbitrary] so the results are deterministic.